mlflow/spark.py

```diff
@@ -895,13 +898,16 @@ class _PyFuncModelWrapper:
     Wrapper around Spark MLlib PipelineModel providing interface for scoring pandas DataFrame.
     """
 
-    def __init__(self, spark, spark_model):
+    def __init__(self, spark, spark_model, signature=None):
         self.spark = spark
         self.spark_model = spark_model
+        self.signature = signature
 
     def predict(self, pandas_df):
         """
         Generate predictions given input data in a pandas DataFrame.
+        If a signature with named model outputs has been defined these outputs will be stored
```

Comment:
I don't think 'stored' is the right word here. 'generated'  or 'produced' might work better.

```suggestion
        If the Spark model has been saved with a signature, the output column names defined during signature creation will be included in the output schema from a `predict()` transformation. If no signature is present or if the model's output would otherwise produce only an output column named 'prediction', the output from this method will contain only a 'prediction' field.
```